app/services/user_event_creator.rb

```diff
@@ -24,23 +30,28 @@ def create_out_of_band_user_event(event_type)
   end
 
   def create_user_event_with_disavowal(event_type, user = current_user)
-    event = create_user_event(event_type, user)
-    EventDisavowal::GenerateDisavowalToken.new(event).call
-    event
+    disavowal_token = SecureRandom.urlsafe_base64(32)
+    create_user_event(event_type, user, disavowal_token)
```

Comment:
maybe could consider switching to keyword args and just inlining?
```suggestion
    create_user_event(event_type, user, disavowal_token: SecureRandom.urlsafe_base64(32))
```